iothub/device/src/Transport/Mqtt/MqttTransportHandler.cs

```diff
@@ -217,12 +217,14 @@ internal MqttTransportHandler(PipelineContext context, IotHubClientMqttSettings
             }
 
             var tlsParameters = new MqttClientOptionsBuilderTlsParameters();
-
-            List<X509Certificate> certs = _connectionCredentials.ClientCertificate == null
-                ? new List<X509Certificate>(0)
-                : new List<X509Certificate> { _connectionCredentials.ClientCertificate };
-
-            tlsParameters.Certificates = certs;
+            if (_connectionCredentials.ClientCertificate != null)
+            {
+                tlsParameters.Certificates = new List<X509Certificate> { _connectionCredentials.ClientCertificate };
+            }
+            else
+            {
+                tlsParameters.IgnoreCertificateRevocationErrors = true;
```

Comment:
Is there a spec that defines the relationship between client certificates being set and the certificate revocation check being set in TLS settings? If not, then can we expose this setting to the user so that they can set it however they want. That way we can address issues such as https://github.com/Azure/azure-iot-sdk-csharp/issues/2921.

Reply:
Yeah, this seems like a bad assumption to me.

Reply:
Now using the 'CertificateRevocationCheck' in IotHubClientTransportSettings to control it.